### PR TITLE
Olelensmar/feat/owner refs

### PR DESCRIPTION
**Pull request by olensmar**
This PR implements the core part of https://github.com/kubeshop/monokle/issues/2036 by adding three new ResourceRef types - incomingOwner, outgoingOwner, unsatisfiedOutgoingOwner and a corresponding link-validation rule to the links validator - enabled by default.

## Changes

-

## Fixes

-

## Checklist

- [X] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [X] added a test

### CODE REVIEW

**Review by WitoDelnat — APPROVED** [review]
LGTM

File: packages/validation/src/references/utils/getResourceNodes.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import {
   ResourceRefsProcessingConfig,
 } from "../../common/types.js";
 import { REF_PATH_SEPARATOR, NAME_REFNODE_PATH } from "../../constants.js";
+import isDeepEqual from 'fast-deep-equal/es6/react';
```
Comment (WitoDelnat) [inline packages/validation/src/references/utils/getResourceNodes.ts:9]: Any reason why you use the React version? Seems a bit out of place as this is unrelated to frontend components.

Comment (olensmar) [reply packages/validation/src/references/utils/getResourceNodes.ts]: ok - updated!

**Review by WitoDelnat — APPROVED** [review]
